packages/core/src/core/logger.test.ts

```diff
@@ -394,15 +394,27 @@ describe('Logger', () => {
     ];
 
     it.each([
-      { tag: 'test-tag', sanitizedTag: 'test-tag' },
-      { tag: 'invalid/?*!', sanitizedTag: 'invalid' },
-      { tag: '/?*!', sanitizedTag: 'default' },
-      { tag: '../../secret', sanitizedTag: 'secret' },
-    ])('should save a checkpoint', async ({ tag, sanitizedTag }) => {
+      {
+        tag: 'test-tag',
+        encodedTag: encode('test-tag'),
```

Comment:
do not call encode here. Include what it encodes to. otherwise someone could break your encode method and your tests would still pass.